### DISCUSSION TITLE
Tenant's notifications do not run on queue

### DISCUSSION THREAD

**Discussion by m7vm7v** [spatie/laravel-multitenancy#441] [Q&A]
I've noticed a few issues here on that but couldn't find my case, apologies if that has been already answered.

I have Event, Listener(ShouldQueue) and Notification(ShouldQueue) -> for the tenants.
I have a Job for the landlord.

When I run the `php artisan queue:work` it executes the landlord's Job and nothing else.
If I try to run the `php artisan tenants:artisan "queue:work" --tenant={the id}` it shows that executes for the tenant but the job remains in the database unexecuted.

I have tried adding `TenantAware` to both listener and notification (they did get tenantId in the job), tried adding both to the `tenant_aware_jobs` and to `queueable_to_job => as 'notification' //each` and also the `queues_are_tenant_aware_by_default => true`. I know all these are supposed to be resolving the same but none of them let me execute the tenant's jobs.

Can you please advise?

**Comment by masterix21** [comment]
What driver are you using?

**Reply by m7vm7v** [reply]
It is the database driver.

**Comment by masterix21** [comment]
What's the place of your tenant jobs? Landlord database?

**Reply by m7vm7v** [reply]
No, the landlord database has a jobs table, that is executed all correctly. Each tenant has its own jobs table, that isn't executed at all.

**Reply by masterix21** [reply]
What do you think about moving all your jobs to the landlord database? Then, starting only one `queue:work` you could optimize the jobs: it's my approach all the time. Give me feedback.

**Reply by m7vm7v** [reply]
Thanks @masterix21 , that is an option but will require a lot of code to be rewritten in my application. Is that the intended way for the package? I was under the impression we separate everything, including the jobs per tenant. I do see the benefit of running just one worker to rule them all, but prefer the separation in the data. In case I inherit your approach how would you fire the event to the landlord's DB? I have the `event(new NewDataRequestSent([data...]));` and currently creates the job for the tenant with `tenantId` inside the tenant's database.

**Reply by m7vm7v** [reply]
Tried 
```
        Landlord::execute(function () {
            event(new NewDataRequestSent(data...));
        });
```
but it didn't work.

**Comment by m7vm7v** [comment]
Can you please advise how to send a tenant-aware event to the landlord's database?

When I dispatch the event in the pure form it does save it in the database for the tenant (including the tenantId in the payload).
If I wrap the same call with the landlord's execute method, it unsets the tenant, dispatches the event and the tenantId is null, reverts the tenant and hits 2 exceptions.

`No 'tenantId' was set in the payload.` and `Call to a member function prepare() on null`. Although, even if is hardcode the tenantId it still hits the second exception. Tried to set the `SESSION_DRIVER = file` and after clearing the cache it still hits the exception.

Any advise is appreciated.

**Reply by m7vm7v** [reply]
In case anyone is wondering you can set the queue database connection to landlord directly in queue.php

```
        'database' => [
            'connection' => 'landlord',
            'driver' => 'database',
            'table' => 'jobs',
            'queue' => 'default',
            'retry_after' => 90,
            'after_commit' => false,
        ],
```
